### PR TITLE
Have .more messages be tagged

### LINKED ISSUE #292: Have .more messages be (optionally) tagged
See https://github.com/goavki/phenny/issues/259#issuecomment-354166715 for reference.

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #292
@wei2912 /cc
@shardulc /cc

### CODE REVIEW

**Comment by gstark0** [comment]
@Androbin take a look at chat

**Comment by sushain97** [comment]
It seems like it would be possible to have `more.add_messages` use the caller's file name by default and then have a `more.add_tagged_messages` that allows specifying the message. This would prevent all the duplication. Or even, use `functools.partial` to freeze the function at the start of each file that uses it often.

**Comment by Androbin** [comment]
@sushain97 Caller sensitivity makes things break all too easily.
And `functools.partial` does not cut more redundancy than it adds.

**Comment by sushain97** [comment]
Why?

**Comment by Androbin** [comment]
* Caller (-> Proxy) -> Function [breaks]
* Caller -> Function (-> Proxy) [breaks]

**Comment by Androbin** [comment]
* An extra line just to add the parameter
* Two different, yet identical functions in the same file
* Any changes need to be applied to both versions

If we were to use `functools.partial` for all such things, we would end up with hundreds of one-line proxy functions.

**Comment by shardulc** [comment]
What about using [this](https://docs.python.org/3/library/inspect.html#inspect.stack) to use the calling function/module name by default?

**Comment by Androbin** [comment]
@shardulc See the suggestion by @sushain97 and my response above.

**Comment by shardulc** [comment]
What breaks, exactly? I'm suggesting that if `tag` is not provided to `more.add_messages`, then it simply inspects the stack and uses the function/module name as the default tag.

**Comment by Androbin** [comment]
@shardulc See https://github.com/goavki/phenny/pull/352#issuecomment-357004967

**Comment by shardulc** [comment]
I read that and I did not understand what you are talking about. What breaks?

**Comment by Androbin** [comment]
The callstack can have a proxy as the previous element instead of the function/module we want.

**Comment by shardulc** [comment]
I agree that proxies can be an issue if we use the function name, so let's not do that. What about using the filename? If there's any situation where a module calls a function in another module entirely to add something to the message queue, that's bad design, so it's safe to use this approach?

**Comment by Androbin** [comment]
> so it's safe to use this approach?

I would rather not use it, as a matter of principle.
But we might as well use it for now, so I will implement it.

**Comment by Androbin** [comment]
Should be fine.

**Comment by Androbin** [comment]
@shardulc Are you okay with this?

File: modules/more.py [hunk 1]
```diff
@@ -31,6 +32,10 @@ def add_messages(target, phenny, messages):
 
     messages = sum(map(lambda message: break_up(message), messages), [])
 
+    if not tag:
+        tag = calling_module()
+        print(tag)
```
Comment (shardulc) [inline modules/more.py]: :smiley:

**Comment by shardulc** [comment]
Looks good.

**Pull request merged**